src/distribution.cpp

```diff
@@ -41,6 +41,19 @@ Discrete::Discrete(const double* x, const double* p, int n)
   this->init_alias(x_vec, p_vec);
 }
 
+Discrete::Discrete(const double* p, int n)
+{
+  std::vector<double> p_vec(p, p + n);
+  std::vector<double> x_vec(n);
+
+  for (int i=0; i<n; i++) 
+  {
+    x_vec[i] = i;
+  }
+
+  this->init_alias(x_vec, p_vec);
+}
+
 void Discrete::init_alias(vector<double>& x, vector<double>& p)
 {
   x_ = x;
```

Comment:
It would be nice to avoid using extra memory by forming the temporary x vector here. Could you make init_alias take an rvalue reference so that we don't have to have two duplicates of this vector in memory at the same time? Then you'd do `x_ = std::move(x)`, I believe. It would reduce memory use for the case of very large meshes. Same for `std::move` usage when calling `init_alias` in the new Discrete constructor you wrote.

Reply:
Hmmm…. All of these constructors make a temporary vector and then only use it to immediately copy to another vector - I’ll do something about that.

Reply:
That would be great; it should be pretty easy to fix this up. Hope this isn't PR project bloat.

Reply:
no bloat at all... just a few minutes to improve (assuming it works ... the 2nd time)